nodejs12.x/cookiecutter-typescript-app-template/{{cookiecutter.project_name}}/app/src/apps/post-app.ts

```diff
@@ -0,0 +1,47 @@
+import { ApiGatewayEvent } from '../common/apigateway/apigateway-event';
+import { ApiGatewayResponse } from '../common/apigateway/apigateway-response';
+import { TodoRepository } from '../common/todo-repository';
+import { TodoItem } from '../common/todo-item';
+
+import { LambdaApp } from './lambda-app';
+
+/**
+ * PostApp is a LambdaApp that puts a new record into DynamoDB using the API Gateway event body as the record content.
+ * 
+ */
+export class PostApp implements LambdaApp {
+    table: string;
+    repository: TodoRepository;
+    
+    constructor(table: string, repository: TodoRepository) {
+        this.table = table;
+        this.repository = repository;
+    }
+    
+    async run(event: ApiGatewayEvent): Promise<ApiGatewayResponse> {
+        let todo: TodoItem
+        try {
+            todo = JSON.parse(event.body);
+            if (!todo.title) {
+                console.log('Body is missing the title');
+                return { statusCode: 422 };
+            } else if (!todo.isComplete) {
+                todo.isComplete = false;
+            } else if (!todo.id) {
+                console.log('Body is missing the id');
+                return { statusCode: 422 };
+            }
+        } catch (err) {
```

Comment:
You're catching JSON parse failures as a 400 (which seems valid), but your tests expect a 500 error. I'd change the test.